OpenRA.Mods.Common/Traits/Buildings/Gate.cs

```diff
@@ -109,15 +112,14 @@ void INotifyBlockingMove.OnNotifyBlockingMove(Actor self, Actor blocking)
 				desiredPosition = OpenPosition;
 		}
 
-		bool CanRemoveBlockage(Actor self, Actor blocking)
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
 		{
-			return !self.IsDisabled() && BuildComplete && blocking.AppearsFriendlyTo(self);
+			blockedPositions = Footprint;
 		}
 
-		protected override void AddedToWorld(Actor self)
+		bool CanRemoveBlockage(Actor self, Actor blocking)
 		{
-			base.AddedToWorld(self);
-			blockedPositions = Info.Tiles(self.Location);
+			return !IsTraitDisabled && !IsTraitPaused && building.BuildComplete && blocking.AppearsFriendlyTo(self);
```

Comment:
This seems to be missing a `&& !building.Locked` check to be consistent with the one in `Tick()`.